### ISSUE TITLE
Make RrecordTypeInfoIdDeveloperName sane again

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#2139]
**[Mingle Card: 2380](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/2380)**
Narrative
---------

There are about half a dozen ways we sync down, pass around, resolve, and set record type values that's causing a whole lot of brittleness scattered throughout the system.

Acceptance Criteria

Essentially, anything that relies on record type values should be verified that they still work. Here's a list of what I can think of at the moment:

1.  Record Type value in any layout (normal form, compact layout, search layout)
2.  Record Type options when creating a job/ticket
3.  Record Type options when adding ticket item
4.  Record type is used to fetch matching layout when editing job/ticket
5.  Record type is used to match layouts for each row in ticket items grid
6.  Workflow status changes
7.  Checklists
8.  Reports (to determine report templates by record type)
9.  Adding ticket items (for each type of item: Quoted item, equipment, scheduled, etc)

Basically....everything hinges on recordtype to work, and it's the most inconsistent piece of data that's tracked in the system...

Related Stories

Tasks
-----

{{table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD}}

Defects
-------

{{table query: SELECT Number, Name, Owner, 'Defect Status' WHERE Type = Defect and 'Related Story' = THIS CARD}}

Test Plan
---------
